src/RequestExecutor/Request.spec.ts

```diff
@@ -132,5 +132,55 @@ describe('Request', () => {
         host: 'example.com, example1.com'
       });
     });
+
+    it('should load different certs per matching port', async () => {
```

Comment:
Pls add at least one complex test case for cert with empty port (default port for https, for instance):
```
      const cert1 = {
        path: '~/cert1.pfx',
        hostname: 'foo.bar',
        port: '8080'
      };

      const cert2 = {
        path: '~/cert2.pfx',
        hostname: 'foo.bar'
      };
```